pkg/collectsub/datasource/filesource/filesource.go

```diff
@@ -0,0 +1,132 @@
+//
+// Copyright 2023 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filesource
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/guacsec/guac/pkg/collectsub/datasource"
+	"gopkg.in/yaml.v3"
+)
+
+type fileDataSources struct {
+	filePath string
+}
+
+type FileFormat struct {
+	OciDataSources []string `yaml:"oci"`
+	GitDataSources []string `yaml:"git"`
+}
+
+/*
+
+NewFileDataSources creates a datasource which gets its data sources
+from a configuration file. This configuration file is in YAML and
+follows the sturcture outlined in the FileFormat struct. An example
+is as follows:
+
+sources.yaml
+----
+oci:
```

Comment:
Yes! It will be need to be documented! It should end up being part of the collector documentation... which I know in general documentation is a bit lacking :(... Once we have an example on an integration, we will need to write up a "How to write a collector" tutorial which will allow help folks to write their own collector implementations! 

Right now it has no consumers yet, so the description just lives in the doc comments.

Reply:
Ok, thanks! 👍